test/utils/set-deep.test.ts

```diff
@@ -228,6 +228,112 @@ describe('Unit | Utility | set deep', () => {
     });
   });
 
+  it('set on nested properties within an empty object', () => {
+    const objA = {
+      top: {}
+    };
+    let value = setDeep(objA, 'top.name', new Change('zoo'));
+
+    expect(value).toEqual({
+      top: {
+        name: new Change('zoo')
+      }
+    });
+
+    value = setDeep(value, 'top.foo.other', new Change('baz'));
+
+    expect(value).toEqual({
+      top: {
+        name: new Change('zoo'),
+        foo: {
+          other: new Change('baz')
+        }
+      }
+    });
+  });
+
+  describe('arrays at various nestings within an object', () => {
```

Comment:
I've started using `describe` blocks for groups of this stuff so I can debug scoped down sections of the test suite, so:
```bash
yarn test:debug:named "arrays at various nestings within an object"
```
allows me to debug just this subset of tests